src/intel/RoomIntel.ts

```diff
@@ -456,7 +458,46 @@ export class RoomIntel {
 			}
 		}
 	}
+	private static autoPoison(room: Room) {
+		
+		if(!Memory.settings.autoPoison.enabled || !room || getAllColonies().length == Game.gcl.level) {
+			return;
+		}
 
+		// poison X rooms at a time
+		let count = 0;
+		for (const colony of getAllColonies()) {
+			if(Game.flags['POISON-'+colony.name]) {
+				count++;
+			}
+			if (count >= Memory.settings.autoPoison.concurrent) {
+				return;
+			}
+		}
```

Comment:
you can use _.filter or _.count
Also, why like this? Naming the poison after the colony name seems bad, better to name it poison-roomyouwanttopoison, otherwise a room can only run 1 at a time.